From 99818dca2d98cff51a18a074aad3dcfaec960c18 Mon Sep 17 00:00:00 2001
From: tchalivendra <tchalivendra@maxlinear.com>
Date: Mon, 12 Aug 2024 16:24:12 +0530
Subject: [PATCH] WLANRTSYS-81666 After reconfiguring hidden SSID,STA unable to
 connect

Issue:
When we reconfigure ssid ,the old ssid is getting appended to new ssid if
we give less number of characters than old one.This causing problem
in RNR short ssid update. So sta is unable to connect if we reconfigure ssid.

Fix:
Clearing old ssid before copying new one.

Signed-off-by: Tejaswi tchalivendra@maxlinear.com
---
 hostapd/config_file.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/hostapd/config_file.c b/hostapd/config_file.c
index 826ac0dcc..cd8545957 100644
--- a/hostapd/config_file.c
+++ b/hostapd/config_file.c
@@ -3045,6 +3045,9 @@ static int hostapd_config_fill(struct hostapd_config *conf,
 				   line, pos);
 			return 1;
 		}
+#ifdef CONFIG_VENDOR_MXL
+		os_memset(ssid->ssid, 0, SSID_MAX_LEN);
+#endif /* CONFIG_VENDOR_MXL */
 		os_memcpy(ssid->ssid, pos, ssid->ssid_len);
 		ssid->ssid_set = 1;
 		ssid->short_ssid = ieee80211_crc32(ssid->ssid, ssid->ssid_len);
-- 
2.43.0

